python/pyspark/sql/udf.py

```diff
@@ -172,5 +179,5 @@ def asNondeterministic(self):
 
         .. versionadded:: 2.3
         """
-        self._deterministic = False
+        self.deterministic = False
```

Comment:
Can we call it `udfDeterministic` to be consistent with Scala side?

https://github.com/apache/spark/blob/ff48b1b338241039a7189e7a3c04333b1256fdb3/sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala#L33

The opposite works fine to me too.

Reply:
`deterministic` is used in `UserDefinedFunction.scala`. Users can use it to check whether this UDF is deterministic or not.